changelogs/unreleased/gh-8576-set-compat-options-to-new.md

```diff
@@ -0,0 +1,11 @@
+## feature/compat
+
+* The following compatibility options' defaults were switched to new behavior:
+    * `yaml_pretty_multiline`
+    * `sql_seq_scan_default`
+    * `json_escape_forward_slash`
+    * `fiber_channel_close_mode`
+    * `fiber_slice_default`
+    * `box_cfg_replication_sync_timeout`
```

Comment:
failed to found a patch for switching default behavior for `box_cfg_replication_sync_timeout`

Reply:
It is in `src/box/lua/load_cfg.lua`:
```diff
 local compat_options = {
     {
         name = 'replication_sync_timeout',
         brief = replication_sync_timeout_brief,
         oldval = 300,
         newval = 0,
         obsolete = nil,
-        default = 'old',
+        default = 'new',
     },
 }

Reply:
Why it is not covered by tests?

Reply:
It is (somewhat): `test/box-luatest/gh_7988_auth_type_test.lua`.
I couldn't update any other tests, because non were [added with the option](https://github.com/tarantool/tarantool/commit/9e42f0c5b454cb0f8838aaefacba3733e6d87012).

Reply:
It is funny. Original author missed testing of introduced compat option, you change behavior and miss testing too.
As a result no one tests this option at all neither "old" nor "new" behavior.

There are many tests, that uses this timeout:

```
[0] ~/sources/MRG/tarantool$ grep -R replication_sync_timeout test/replication
test/replication/wal_off.test.lua:replication_sync_timeout = box.cfg.replication_sync_timeout
test/replication/wal_off.test.lua:box.cfg { replication_sync_timeout = 0.01 }
test/replication/wal_off.test.lua:box.cfg { replication_sync_timeout = replication_sync_timeout }
test/replication/replica_uuid_rw.lua:    replication_sync_timeout = arg[4] and tonumber(arg[4]) or 1.0,
test/replication/replica_uuid_rw2.lua:    replication_sync_timeout = arg[4] and tonumber(arg[4]) or 1.0,
test/replication/replica_uuid_rw3.lua:    replication_sync_timeout = arg[4] and tonumber(arg[4]) or 1.0,
test/replication/wal_off.result:replication_sync_timeout = box.cfg.replication_sync_timeout
test/replication/wal_off.result:box.cfg { replication_sync_timeout = 0.01 }
test/replication/wal_off.result:box.cfg { replication_sync_timeout = replication_sync_timeout }
test/replication/sync.test.lua:box.cfg{replication_sync_lag = 0.001, replication_sync_timeout = 0.001}
test/replication/sync.test.lua:box.cfg{replication_sync_timeout = 10}
test/replication/gh-5445-leader-inconsistency.result:-- Otherwise it would stall for replication_sync_timeout. This is due to the
test/replication/sync.result:box.cfg{replication_sync_lag = 0.001, replication_sync_timeout = 0.001}
test/replication/sync.result:box.cfg{replication_sync_timeout = 10}
test/replication/gh-5445-leader-inconsistency.test.lua:-- Otherwise it would stall for replication_sync_timeout. This is due to the
test/replication/replica_uuid_rw1.lua:    replication_sync_timeout = arg[4] and tonumber(arg[4]) or 1.0,
```

I propose to add testcases for replication with default sync replication timeout for a new and old behaviours, like we did for other tests.

Reply:
Can you create an issue for that? I think it is out of scope for this patch.

Reply:
Cherry-picked relevant replication test changes.

Reply:
Thanks!